submission/submission-core/src/main/kotlin/ac/uk/ebi/biostd/submission/domain/request/SubmissionRequestCleanIndexer.kt

```diff
@@ -43,7 +43,7 @@ class SubmissionRequestCleanIndexer(
         processId: String,
     ): SubmissionRequest =
         requestService.onRequest(accNo, version, RequestStatus.LOADED, processId) {
-            val indexedRequest = indexRequest(it.process.submission)
+            val indexedRequest = indexRequest(it.process!!.submission)
```

Comment:
there is something missing in the modeling IMO that we need to use so much the it.process!!, I am afraid I can not see it now but we can take a look and improve after the completion of this feature

Reply:
I completely agree. Ideally we should be able to have a class with and without without too much mapping